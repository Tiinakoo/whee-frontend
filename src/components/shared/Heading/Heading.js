import React from "react";
import upperFirst from "lodash/fp/upperFirst";
import styles from "./Heading.module.scss";
import classNames from "classnames";

const Heading = ({
  size = "md",
  centeredVertically,
  className,
  children,
  ...props
}) => {
  const classes = classNames([
    styles.heading,
    styles[`heading__size${upperFirst(size)}`],
    {
      [styles.heading__centeredVertically]: centeredVertically,
    },
    className,
  ]);

  return (
    <p className={classes} {...props}>
      {children}
    </p>
  );
};

export default Heading;
