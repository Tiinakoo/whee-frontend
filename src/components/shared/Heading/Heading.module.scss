@import "../../../styles/mixins";
@import "../../../styles/baseline";
@import "../../../styles/colors";
@import "../../../styles/typography";

.heading {
  margin: 0;
  color: $secondaryColor;
  @include font-heading();

  &__alignRight {
    text-align: right;
  }

  &__sizeSm {
    font-size: 8 * $typographyBaseline;
  }

  &__sizeMd {
    font-size: 10 * $typographyBaseline;

    @include md {
      font-size: 12 * $typographyBaseline;
    }

    @include lg {
      font-size: 14 * $typographyBaseline;
    }
  }

  &__sizeLg {
    font-size: 12 * $typographyBaseline;

    @include md {
      font-size: 14 * $typographyBaseline;
    }
  }

  &__centeredVertically {
    align-self: center;
  }
}
