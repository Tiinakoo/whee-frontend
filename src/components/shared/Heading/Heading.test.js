import { mount } from "enzyme";
import React from "react";

import Heading from "./Heading";

describe("Heading", () => {
  it("renders", () => {
    const component = mount(<Heading>Some button</Heading>);

    expect(component).toMatchSnapshot();
  });
});
