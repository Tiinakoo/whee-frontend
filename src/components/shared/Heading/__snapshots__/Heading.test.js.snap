// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders 1`] = `
<Heading>
  <p
    className="heading heading__sizeMd"
  >
    Some button
  </p>
</Heading>
`;
